hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/JdbcSource.java

```diff
@@ -254,22 +257,23 @@ private Dataset<Row> fullFetch(long sourceLimit) {
     return validatePropsAndGetDataFrameReader(sparkSession, props).option(Config.RDBMS_TABLE_PROP, query).load();
   }
 
-  private String checkpoint(Dataset<Row> rowDataset, boolean isIncremental, Option<String> lastCkptStr) {
+  private Checkpoint checkpoint(Dataset<Row> rowDataset, boolean isIncremental, Option<Checkpoint> lastCheckpoint) {
     try {
       if (isIncremental) {
         Column incrementalColumn = rowDataset.col(getStringWithAltKeys(props, JdbcSourceConfig.INCREMENTAL_COLUMN));
         final String max = rowDataset.agg(functions.max(incrementalColumn).cast(DataTypes.StringType)).first().getString(0);
         LOG.info(String.format("Checkpointing column %s with value: %s ", incrementalColumn, max));
         if (max != null) {
-          return max;
+          return new CheckpointV2(max);
```

Comment:
can we introduce a factory.. `Checkpoint.create(..)` that will create the current/latest by default.. then we avoid use of version specific `CheckpointV2` and CheckpointV1` classes in regular code.. There should be a few places where we will handle CheckpointV1 and thats it